src/flightRecorder.cpp

```diff
@@ -198,6 +200,9 @@ class Lookup {
             mi->_sig = _symbols.lookup("()L;");
         }
 
+        if (method_class) {
+            _jni->DeleteLocalRef(method_class);
```

Comment:
Hi,

After investigating why the call to `DeleteLocalRef` is added here(cannot completely fix the problem), I think we can fix the problem in OpenJDK upstream.

Do you plan to fix it? If not, we will file an issue at OpenJDK.

Reply:
I have created a backport into 17 for JDK-8268364 at https://github.com/openjdk/jdk17u-dev/pull/2839. Do you propose a different fix?

Reply:
> Do you propose a different fix?

If I understand correctly, the call to 'DeleteLocalRef' here to reduce the possibility of a crash. That crash is actually a bug in JDK and not fixed by JDK-8268364. So we should file a new issue in OpenJDK to fix it.

Reply:
@D-D-H You are right: `DeleteLocalRef` is a cheap way to significantly decrease possibility of a crash, but fixing the root cause in OpenJDK is still highly desirable. I think it makes sense to integrate this change anyway, before the proper fix is released.

Reply:
My colleague Liang created an issue for this problem: https://bugs.openjdk.org/browse/JDK-8339725

Reply:
Cool, thanks!